### PR TITLE
improve version readability

### PR DESCRIPTION
**Pull request by vprashar2929**

### CODE REVIEW

File: .github/workflows/image.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
         - name: Generate Version
           id: version
           run: |
-            version="$(cat VERSION)-$(date +%y%m%d%H%M%S)"
+            version="$(cat VERSION)-$(date +%Y-%m-%d-%H:%M:%S)"
```
Comment (sthaha) [inline .github/workflows/image.yaml:29]: Have you tried making images using this and also tried bundle run? 
Could you please point to a CI run? 

I think all non-trivial CI (especially those that are run after merge) changes must be tested in a fork

**Review by sthaha — CHANGES_REQUESTED** [review]

File: .github/workflows/image.yaml (hunk 1 above)
Comment (vprashar2929) [reply .github/workflows/image.yaml]: Absolutely I agree🙂. It is best that we highlight this in the Developer Docs section.

**Comment by vprashar2929** [comment]
Test CI run: https://github.com/vprashar2929/kepler-operator/actions/runs/5992617684/job/16252190833

**Comment by sthaha** [comment]
> Test CI run: https://github.com/vprashar2929/kepler-operator/actions/runs/5992617684/job/16252190833

@vprashar2929 I see 
```
docker build -f bundle.Dockerfile \
	-t quay.io/sustainable_computing_io/kepler-operator-bundle:0.6.2-20230827184303 \
	--platform=linux/amd64 .
```
https://github.com/vprashar2929/kepler-operator/actions/runs/5992617684/job/16252190833#step:6:52

Which isn't this change, is it?

**Comment by sthaha** [comment]
@vprashar2929 I was also looking for the bundle image, so that we can try `operator-sdk run bundle`. There are some quarks with operator bundle and versioning which is why I am asking if you were able to use operator-sdk run bundle.

**Comment by sthaha** [comment]
I have added https://github.com/sustainable-computing-io/kepler-operator/pull/180 to see if the versioning works.

**Comment by sthaha** [comment]
@vprashar2929 As I suspected, 
1. Docker tag is invalid - see: https://github.com/sustainable-computing-io/kepler-operator/actions/runs/5993765897/job/16254528597?pr=180#step:7:73

```
docker build -t localhost:5001/kepler-operator/kepler-operator:0.6.2-2023-08-28-00:06:35 \
--build-arg TARGETOS=linux \
--build-arg TARGETARCH=amd64 \
--platform=linux/amd64 .
ERROR: invalid tag "localhost:5001/kepler-operator/kepler-operator:0.6.2-2023-08-28-00:06:35": invalid reference format
make: *** [Makefile:127: operator-build] Error 1
Error: Process completed with exit code 2.
```

3. I don't think `operator-sdk run bundle` would have worked as well.

**Comment by vprashar2929** [comment]
>Which isn't this change, is it?

The current change just [updates](https://github.com/sustainable-computing-io/kepler-operator/pull/178/files#) `+%y` to `+%Y`

**Review by sthaha — APPROVED** [review]

**Pull request merged**